cmd/operator-sdk/internal/genutil/openapi.go

```diff
@@ -95,25 +93,35 @@ func OpenAPIGen() error {
 	return nil
 }
 
-func buildOpenAPIGenBinary(binDir, codegenSrcDir string) error {
-	genDirs := []string{"./cmd/openapi-gen"}
-	return buildCodegenBinaries(genDirs, binDir, codegenSrcDir)
-}
-
-func openAPIGen(binDir, hf string, fqApis []string) (err error) {
-	cgPath := filepath.Join(binDir, "openapi-gen")
-	for _, fqApi := range fqApis {
-		args := []string{
-			"--input-dirs", fqApi,
-			"--output-package", fqApi,
-			"--output-file-base", "zz_generated.openapi",
-			// openapi-gen requires a boilerplate file. Either use header or an
-			// empty file if header is empty.
-			"--go-header-file", hf,
+func openAPIGen(hf string, fqApis []string) error {
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	flag.Set("logtostderr", "true")
+	for _, api := range fqApis {
+		// Use relative API path so the generator writes to the correct path.
+		apiPath := "./" + api[strings.Index(api, scaffold.ApisDir):]
+		args := &gengoargs.GeneratorArgs{
+			InputDirs:          []string{apiPath},
+			OutputFileBaseName: "zz_generated.openapi",
+			OutputPackagePath:  filepath.Join(wd, apiPath),
+			GoHeaderFilePath:   hf,
+			CustomArgs: &generatorargs.CustomArgs{
+				ReportFilename: "-", // stdout
```

Comment:
Nit: Make the comment on this option a little more clear
```suggestion
				// Print API rule violations to stdout
				ReportFilename: "-",
```